src/Compilers/Core/MSBuildTaskTests/CscTests.cs

```diff
@@ -91,6 +91,20 @@ public void RuntimeMetadataVersionFlag()
         }
 
         [Fact]
+        public void ChecksumAlgorithmFlag()
+        {
+            var csc = new Csc();
+            csc.Sources = MSBuildUtil.CreateTaskItems("test.cs");
+            csc.ChecksumAlgorithm = "sha256";
+            Assert.Equal("/out:test.exe /checksumalgorithm:sha256 test.cs", csc.GenerateResponseFileContents());
+
+            csc = new Csc();
+            csc.Sources = MSBuildUtil.CreateTaskItems("test.cs");
+            csc.ChecksumAlgorithm = null;
```

Comment:
I'll add a test, but this is not representable in MSBuild. Passing an empty value to the property of  a task is the same as assigning its default value (null).